app/controllers/admin/domain_versions_controller.rb

```diff
@@ -1,77 +1,69 @@
 module Admin
   class DomainVersionsController < BaseController
     include ObjectVersionsHelper
+    include Auditable
 
-    load_and_authorize_resource
+    load_and_authorize_resource class: 'Audit::DomainHistory'
 
     def index
       params[:q] ||= {}
 
-      @q = DomainVersion.includes(:item).search(params[:q])
+      @q = Audit::DomainHistory.search(params[:q])
       @versions = @q.result.page(params[:page])
       search_params = params[:q].deep_dup
 
       if search_params[:registrant].present?
-        registrants = Contact.where("name ilike ?", "%#{search_params[:registrant].strip}%")
+        registrants = Contact.where('name ilike ?', "%#{search_params[:registrant].strip}%")
         search_params.delete(:registrant)
       end
 
       if search_params[:registrar].present?
-        registrars = Registrar.where("name ilike ?", "%#{search_params[:registrar].strip}%")
+        registrars = Registrar.where('name ilike ?', "%#{search_params[:registrar].strip}%")
         search_params.delete(:registrar)
       end
 
-      whereS = "1=1"
+      whereS = '1=1'
 
       search_params.each do |key, value|
         next if value.empty?
-        case key
-          when 'event'
-            whereS += " AND event = '#{value}'"
-          when 'name'
-            whereS += " AND (object->>'name' ~* '#{value}' OR object_changes->>'name' ~* '#{value}')"
-          else
-            whereS += create_where_string(key, value)
-        end
+
+        whereS += case key
+                  when 'action'
+                    " AND action = '#{value}'"
+                  when 'name'
+                    " AND (new_value->>'name' ~* '#{value}' OR new_value->>'name' ~* '#{value}')"
+                  else
+                    create_where_string(key, value)
+                  end
       end
 
-      whereS += "  AND object->>'registrant_id' IN (#{registrants.map { |r| "'#{r.id.to_s}'" }.join ','})" if registrants.present?
-      whereS += "  AND 1=0" if registrants == []
-      whereS += "  AND object->>'registrar_id' IN (#{registrars.map { |r| "'#{r.id.to_s}'" }.join ','})" if registrars.present?
-      whereS += "  AND 1=0" if registrars == []
+      whereS += add_query(contacts: registrants, field: 'registrant_id') if registrants.present?
+      whereS += '  AND 1=0' if registrants == []
+      whereS += add_query(contacts: registrars, field: 'registrar_id') if registrars.present?
+      whereS += '  AND 1=0' if registrars == []
```

Comment:
I know that `whereS` was used before as well, but isn't it wiser to convert it to snake_case as per standard?